src/couch_log/src/couch_log_writer_journald.erl

```diff
@@ -0,0 +1,69 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_log_writer_journald).
+-behaviour(couch_log_writer).
+
+
+-export([
+    init/0,
+    terminate/2,
+    write/2
+]).
+
+
+-include("couch_log.hrl").
+
+
+init() ->
+    {ok, nil}.
+
+
+terminate(_, _St) ->
+    ok.
+
+
+% log level mapping from sd-daemon(3)
+% https://www.freedesktop.org/software/systemd/man/sd-daemon.html
+-spec level_to_integer(atom() | string() | integer()) -> integer().
+level_to_integer(debug)                 -> 7;
+level_to_integer(info)                  -> 6;
+level_to_integer(notice)                -> 5;
+level_to_integer(warning)               -> 4;
+level_to_integer(warn)                  -> 4;
+level_to_integer(error)                 -> 3;
+level_to_integer(err)                   -> 3;
+level_to_integer(critical)              -> 2;
+level_to_integer(crit)                  -> 2;
+level_to_integer(alert)                 -> 1;
+level_to_integer(emergency)             -> 0;
+level_to_integer(emerg)                 -> 0;
+level_to_integer(none)                  -> 6.
```

Comment:
this mapping doesn’t seem right, it should result in NO logging, not info logging.

Reply:
thanks, I'll have to double check that, it's been awhile since I wrote this patch.

Reply:
the `level_to_integer` function here is an internal one that decides how to map the couchdb log levels (as atoms) to the special `<N>` format recognized by journald. 

Further, it's actually `couch_log:log` (that calls `couch_log_util:should_log`) that decides if it should log the event based on the level. From what I see the writer itself doesn't decide on what it should log based on the level (looking at `couch_log_writer_syslog/file/stderr`).

Perhaps, to avoid confusion, I should rename this function to `level_for_journald`?

Reply:
To add to the above, after looking at couch_log.erl (and couch_log_writer_syslog.erl) again, the only exported log functions are: `debug`, `info`, `notice`, `warning`, `error`, `critical`, `alert`, `emergency` with the corresponding atoms as level, so I guess I can make a similar function to the `get_level` from writer_syslog.

Reply:
is this better?

```
% log level mapping from sd-daemon(3)
% https://www.freedesktop.org/software/systemd/man/sd-daemon.html
-spec level_for_journald(atom()) -> integer().
level_for_journald(Level) when is_atom(Level) ->
    case Level of
        debug       -> 7;
        info        -> 6;
        notice      -> 5;
        warning     -> 4;
        error       -> 3;
        critical    -> 2;
        alert       -> 1;
        emergency   -> 0;
        _           -> 3
    end.
```

Reply:
rebased and pushed with that ^